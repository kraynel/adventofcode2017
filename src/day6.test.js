const { getCycleNumbers, getCycleLength } = require("./day6");

describe("getCycleNumbers", () => {
  it("should return the correct number of cycles", () => {
    expect(getCycleNumbers("0 2 7 0")).toBe(5);
  });

  it("should return the correct number of cycles", () => {
    expect(
      getCycleNumbers("0 5 10 0 11 14 13 4 11 8 8 7 1 4 12 11")
    ).toMatchSnapshot();
  });
});

describe("getCycleLength", () => {
  it("should return the correct number of cycles", () => {
    expect(getCycleLength("0 2 7 0")).toBe(4);
  });

  it("should return the correct number of cycles", () => {
    expect(
      getCycleLength("0 5 10 0 11 14 13 4 11 8 8 7 1 4 12 11")
    ).toMatchSnapshot();
  });
});
