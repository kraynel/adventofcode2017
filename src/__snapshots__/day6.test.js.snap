// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCycleLength should return the correct number of cycles 1`] = `1695`;

exports[`getCycleNumbers should return the correct number of cycles 1`] = `7864`;
